src/main/java/ch/njol/skript/effects/EffHealth.java

```diff
@@ -34,42 +18,52 @@
 import ch.njol.util.Math2;
 import org.bukkit.entity.Damageable;
 import org.bukkit.event.Event;
+import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
 import org.bukkit.inventory.ItemStack;
-import org.eclipse.jdt.annotation.Nullable;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
 
 @Name("Damage/Heal/Repair")
-@Description("Damage/Heal/Repair an entity, or item.")
+@Description({
+	"Damage, heal, or repair an entity or item.",
+	"We can use damage entity with a fake damage cause in Spigot 1.20.4+"
+})
 @Examples({
 	"damage player by 5 hearts",
+	"damage player by 3 hearts with fake cause fall",
 	"heal the player",
 	"repair tool of player"
 })
 @Since("1.0")
+@RequiredPlugins("Spigot 1.20.4+ (for damage cause)")
 public class EffHealth extends Effect {
-
 	static {
 		Skript.registerEffect(EffHealth.class,
-			"damage %livingentities/itemtypes/slots% by %number% [heart[s]] [with fake cause %-damagecause%]",
+			"damage %livingentities/itemtypes/slots% by %number% [heart[s]] [with [fake] [damage] cause %-damagecause%]",
 			"heal %livingentities% [by %-number% [heart[s]]]",
 			"repair %itemtypes/slots% [by %-number%]");
 	}
 
+	private final boolean canSetDamageCause = Skript.classExists("org.bukkit.damage.DamageSource");
```

Comment:
I think this can be made a static constant, and moved above the static block
```suggestion
	private static final boolean HAS_DAMAGE_SOURCE = Skript.classExists("org.bukkit.damage.DamageSource");
```